### DISCUSSION TITLE
How To Build For PC?

### DISCUSSION THREAD

**Discussion by dmunsie** [airsdk/Adobe-Runtime-Support#3186] [General]
Note: I've just been building Android APKs with the Air SDK so far.

Not for commercial distribution at this point, I just want to build an .exe (or whatever is required) and test it on a few PCs. How would I do this? Thanks.

**Comment by FliplineStudios** [comment]
Is this for packaging a PC application?  (And not just for testing an Android app on a PC?)

For our PC game, we've used `-target bundle` when packaging via ADT:
https://help.adobe.com/en_US/air/build/WSfffb011ac560372f709e16db131e43659b9-8000.html

On Windows, this will create a folder that includes an exe, plus all of your app assets and the captive runtime.

There are a few desktop-specific tags you can add to your application descriptor XML too:

```
<!-- this may be automatically set if you don't specify it -->
<supportedProfiles>desktop extendedDesktop</supportedProfiles>

<!-- prefer discrete GPU if available for better performance, instead of integrated -->
<gpuPreference>discrete</gpuPreference>

<!-- optional, if you want to force DirectX 9 -->
<windows><maxD3D>9</maxD3D></windows>

<!-- optional, if you want to allow users to change stage quality -->
<initialWindow>
  <allowLowQuality>true</allowLowQuality>
</initialWindow>
```

More info about the profiles here:
https://help.adobe.com/en_US/air/build/WS144092a96ffef7cc16ddeea2126bb46b82f-8000.html

Some other PC-specific elements here:
https://airsdk.dev/docs/building/application-descriptor-files/elements/windows

**Comment by atollontech** [comment]
Easiest way is to use some IDE like Intellij IDEA.

In the Intellij IDEA:
1. New Project - Flash, Target: Desktop
2. Choose AIR SDK, Next, Create
3. In the menu Build - Package AIR Application
4. Select AIR, EXE or Captive runtime bundle.
